mumbleBot.py

```diff
@@ -459,6 +505,23 @@ def _download(self, item):
     def loop(self):
         while not self.exit and self.mumble.is_alive():
 
+            # It's posible for libs to start tasks, they are managed/monitored over here.
+            for task in self.task_handle:
+                name = self.task_handle['method']['task_name']
```

Comment:
I happen to be in the middle of considering optimizing the logic of the main loop to make it lighter. And now we have these dict lookups in the middle...

I think if you create a specific class like
```python
from enum import Enum

class TaskState(Enum):
    added = 0
    started = 1

class TaskHandle:
    def __init__(self, name, state, handle):
        self.name = name
        self.state = TaskState.added
        self.handle = handle
    ...
```

However, I'm just wondering why you put task initialization in the main loop?

Reply:
Maybe It's better to just release the loop of already started thread tasks in main, this would also give better control of stopping/pause the task.
I currently placed it in the loop the get them running as late as possible, as they live quite on their own.